src/others/ime/lv_ime_pinyin.c

```diff
@@ -798,6 +798,7 @@ static void pinyin_input_proc(lv_obj_t * obj)
     }
 
     lv_obj_remove_flag(pinyin_ime->cand_panel, LV_OBJ_FLAG_HIDDEN);
+    lv_obj_invalidate(pinyin_ime->cand_panel);
```

Comment:
I see you made it easier to read, at the cost of potentially calling `lv_obj_invalidate` twice :wink:

What if you did this?

```suggestion
    if(lv_obj_has_flag(pinyin_ime->cand_panel, LV_OBJ_FLAG_HIDDEN)) {
        lv_obj_remove_flag(pinyin_ime->cand_panel, LV_OBJ_FLAG_HIDDEN);
    }
    else {
        lv_obj_invalidate(pinyin_ime->cand_panel);
    }
```

Reply:
The invalidation is cheap and if an area is already invalidated it will be filtered out anyway. So I'd keep it as it is for code simplicity.

Reply:
Okay I agree